Update feature (#60)
kyma-project/btp-manager: docs/operations.md

| 12  | Error      | Ready          | False            | ConsistencyCheckFailed | Failure during consistency check                                               |

## Updating

![Updating diagram](./assets/updating.svg)

> 

Reviewer: In the last box in the chart, I guess you mean `orphaned resources`, right?

> The updating logic is based on the chart files placed in the ChartPath.

Reviewer: ```suggestion
The updating logic is based on the chart files placed in the `ChartPath`.
```

| 12  | Error      | Ready          | False            | ConsistencyCheckFailed | Failure during consistency check                                               |

## Updating

![Updating diagram](./assets/updating.svg)

> Updating logic is based on chart files placed in "ChartPath"

Reviewer: ```suggestion
The updating logic is based on the chart files placed in the ChartPath.
```

> Config Map (btp-manager-config-map)

Reviewer 2: Think about different name for ConfigMap. There might be a similar named ConfigMap with configuration for btp-manager:
https://github.com/kyma-project/btp-manager/blob/main/docs/configuration.md

Reviewer 2: how about `btp-manager-versions`?

> 
> Config Map (btp-manager-config-map)

Reviewer: ```suggestion
```

> Application will create config map (if not exists) and then use it for storing data, including: 

Reviewer: ```suggestion
The application creates a ConfigMap, `btp-manager-versions`, if one does not exist,  and then uses it for storing the following data: 
```

> - current version within current installed gvks
> - old version within old version kinds.

Reviewer: ```suggestion
- the current version within currently installed gvks
- the old version within old gvks
```

> All resources will be labeled with label 'app.kubernetes.io/chart-version'.

Reviewer: ```suggestion
The label `app.kubernetes.io/chart-version` is assigned to all resources.
```

> Application after restart, during 1st reconcile loop iteration, inspects Chart.yaml file from which extracts new version of chart.

Reviewer: ```suggestion
After a restart, during the first reconcile loop iteration, the application inspects `Chart.yaml` file from which it extracts a new version of the chart.
```

> If version has changed then it updates config map and shifts value, so it move current values to old values, and new values moves to current values.

Reviewer: ```suggestion
If the version has changed, the application updates the `btp-manager-versions` and shifts values; that is, it moves current values to old values, and new values to current values.
```

> Next step is run to ConsistencyCheck which will handle updated - if there is new resources in charts, it will be applied to cluster, and if something is changed, the resources on cluster will be updated.

Reviewer: ```suggestion
In the next step, call the ConsistencyCheck which handles the update. If there are new resources in the charts, it is applied to the cluster. If some of the chart resources are changed, the resources on the cluster are updated.
```

> The ConsistencyCheck also apply label with current version to all matching resources. 

Reviewer: ```suggestion
The ConsistencyCheck also applies a label with the current version to all matching resources. 
```

> If some resources during check are on cluster and in the same time they are not is not in chart resources, it mean they will stay with old version label.

Reviewer: ```suggestion
If some resources during the check are on the cluster and at the same time they are not in the chart resources, they stay with the old version label.
```

> Resources which are labeled with old version will be deleted by operator.

Reviewer: ```suggestion
Resources labeled with the old version are deleted by the application.
```
